### ISSUE TITLE
Causes subsecond freezes when getting system state

### ISSUE THREAD

**Issue by hyperair** [elvetemedve/gnome-shell-extension-system-monitor#36]
When this extension is enabled, it causes gnome-shell freezes that last for about a second or less at the same frequency as the configured update frequency. This can be seen most easily when watching a continuous animation of some sort, e.g. `glxgears`.

**Comment by elvetemedve** [comment]
Hi @hyperair,

It's a known bug. I have been spent much time making the work of collecting statistics asynchronous to avoid GUI lag. The main problem is that Gnome Shell is single threaded according to the developers which makes things hard.

I think the lag with latest version is acceptable, if not try to turn off the **swap indicator**.

**Comment by hyperair** [comment]
On Thu, Aug 24, 2017 at 06:32:25AM -0700, Géza Búza wrote:
> Hi @hyperair,
> 
> It's a known bug. I have been spent much time making the work of collecting
> statistics asynchronous to avoid GUI lag. The main problem is that Gnome Shell
> is single threaded according to the developers which makes things hard.
> 
> I think the lag with latest version is acceptable, if not try to turn off the
> **swap indicator**.

I see, thanks. That helps a bit. Would it be possible to let the process display
be disabled? It seems like that might be the heaviest portion of the statistics
collection.

-- 
Kind regards,
Loong Jin

**Comment by elvetemedve** [comment]
Unfortunately that would require a lot of work, because each meter collects it's own data regardless of what is displayed by the GUI.
The best would be to do the statistics collection in a separate thread.

**Comment by MatthewStephenson** [comment]
I'm experiencing this too, and have reluctantly disabled the system monitor and gone in search of an alternative, because its effect on video playback is not acceptable to me. I hope this bug can be resolved so I can return to it, because it's really useful.

**Comment by elvetemedve** [comment]
I don't think it will be solved until Gnome 4 ([see 5. pullet point of problem areas](https://wiki.gnome.org/Initiatives/Wayland/GnomeShell/GnomeShell4#Problem_areas)).

**Comment by hyperair** [comment]
On Sun, Jan 28, 2018 at 04:20:57AM -0800, MatthewStephenson wrote:
> I'm experiencing this too, and have reluctantly disabled the system monitor
> and gone in search of an alternative, because its effect on video playback is
> not acceptable to me. I hope this bug can be resolved so I can return to it,
> because it's really useful.

I've switched to Argos[1] with the sysinfo script[2] for now -- all of the heavy
lifting is done out-of-process since Argos is just a simple proxy for
executables' stdout into the panel.


On Sun, Jan 28, 2018 at 02:27:54PM +0000, Géza Búza wrote:
> I don't think it will be solved until Gnome 4 ([see 5. pullet point of problem
> areas](https://wiki.gnome.org/Initiatives/Wayland/GnomeShell/GnomeShell4#Problem_areas)).

Great to know that they're thinking seriously about this. I've had to switch
back to X11 since I've been affected by gnome-shell memory leaks (not sure
whether it's really a leak or just memory fragmentation causing RSS to bloat),
and key-repeat runaways involving Chrome[3]. Being able to restart gnome-shell
without losing all open applications is awesome.

[1] https://extensions.gnome.org/extension/1176/argos/
[2] https://github.com/fadeouter/sysinfo
[3] https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/1731420

-- 
Kind regards,
Loong Jin
